// hhbrown/index.js
var https       = require('https')
  , querystring = require('querystring');

var Api = {};

Api.access_token = null;

Api.connect = function(clientID, clientSecret, callback) {
	var post_data = querystring.stringify({
		grant_type    : 'client_credentials',
		client_id     : clientID,
		client_secret : clientSecret 
	});

  	// An object of options to indicate where to post to
  	var post_options = {
     	host: 'www.apitite.net',
      	port: '443',
      	path: '/api/hhbrown/oauth/access_token',
      	method: 'POST',
      	headers: {
          	'Content-Type': 'application/x-www-form-urlencoded',
          	'Content-Length': post_data.length
      	}
  	};

	// Set up the request
	var post_req = https.request(post_options, function(res) {
		res.setEncoding('utf8');

		res.on('data', function (chunk) {
			Api.access_token = JSON.parse(chunk).access_token_base64;

			post_req.end();
			
			if(callback) callback();
		});
	});

	// post the data
	post_req.write(post_data);
}

Api.get = function(url, params, callback) {
	if(!Api.access_token) return console.log('No Access Token!');

	var newParams = {};

	for(var param in params) {
		if(params[param]) newParams[param] = params[param];
	}

	newParams = querystring.stringify(newParams);

	console.log(newParams)

	var options = {
     	host: 'www.apitite.net',
		port: '443',
		path: '/api/hhbrown/' + url + '/json?' + newParams,
		method: 'GET',
		headers: {
			'Authorization': 'Bearer ' + Api.access_token
		}
	};

	var req = https.get(options, function(res) {
		var data = '';

		res.on('data', function(d) {
			data += d;
		});

		res.on('end', function() {
			callback(JSON.parse(data));
		});
	});
}

exports = module.exports = Api;
